glue_genomics_viewers/genome_track/qt/data_viewer.py

```diff
@@ -69,11 +69,27 @@ def _zoom_to_layer(self):
             self.state.start = start
             self.state.end = end
 
+    def _zoom_to_data_bounds(self, data):
+        bounds = data.get_chrom_bounds()
+        if not bounds:  # empty dataset
+            return
+
+        with delay_callback(self.state, 'chr', 'start', 'end'):
+            chr, start, end = bounds[0]
+            self.state.chr = chr.lstrip('chr')
+            self.state.start = start
+            self.state.end = start + (end - start) / 100
```

Comment:
Coolbox's gene annotation plotter is very slow (and illegible) when trying to render a ~full chromosome, so we default to a smaller chunk.

Reply:
That makes sense. It feels like an interesting summarization problem to show anything meaningful at a wide zoom (density of annotations as a heatmap, perhaps?), but agreed that the approach taken by coolbox does not work for us here.